src/generate/cpp_type.rs

```diff
@@ -230,13 +267,34 @@ impl CppType {
                     .get(field.type_index as usize)
                     .unwrap();
 
+                let cpp_name = self.cpp_name(ctx_collection, metadata, config, f_type, false);
+
                 // Need to include this type
-                let cpp_type_name = ctx.cpp_name(ctx_collection, metadata, config, f_type);
+                let field_cpp_name = self.field_cpp_name(
+                    cpp_name,
+                    *f_offset,
+                    !f_type.is_static() && !f_type.is_const(),
+                    f_type.is_const(),
+                );
 
-                self.declarations.push(Arc::new(CppCommentedString{
-                    data: "".to_string(), // TODO
-                    comment: Some(format!("Field: {i}, name: {f_name}, Type Name: {cpp_type_name}, Offset: {f_offset}"))
+                self.declarations.push(Arc::new(CppCommentedString {
+                    data: format!(
+                        "{} {field_cpp_name} {f_name};",
+                        if f_type.is_static() {
+                            "static inline"
+                        } else {
+                            ""
+                        }
+                    ), // TODO
+                    comment: Some(format!(
+                        "Field: {i}, name: {f_name}, Type Name: {f_type:?}, Offset: 0x{f_offset:x}"
+                    )),
                 }));
+
+                // forward declare only if field type is not the same type as the holder

```

Comment:
Ideally, FDs are specified elsewhere, ex, as part of the context. This is because we need the type impl to hold the includes for the types we actually need for these FDs. As such, we need to keep track (to some degree) of what we FD so that we can include them later. So, ideally, these resolutions go through the context. In fact, all cpp_name resolutions should inherently go through context such that it can determine where the includes/FDs are necessary.

Reply:
Looks like some of this is already happening, but FDs should ideally also be something that is forwarded back to the context so that it can keep track of them. That and/or includes that are necessary on the impl level are also necessary.